lib/features/message/domain/message.dart

```diff
@@ -1,36 +1,16 @@
-/// Message object.
-class Message {
-  /// No set methods as these are all final values
-  final String messageId;
-  final String senderId;
-  final String groupId;
-  final String messageContent;
+import 'package:freezed_annotation/freezed_annotation.dart';
 
-  /// Constructor
-  Message({
-    required this.messageId,
-    required this.senderId,
-    required this.groupId,
-    required this.messageContent,
-  });
+part 'message.freezed.dart';
+part 'message.g.dart';
 
-  /// Get the message ID.
-  String getMessageID() {
-    return messageId;
-  }
+@unfreezed
+class Message with _$Message {
+  factory Message({
+    required final String messageId,
```

Comment:
Also, since all the other models are formatted typeID(instead of typeId), should this be changed to reflect that format?